cmd/sftp-server.go

```diff
@@ -120,6 +125,228 @@ var supportedMACs = []string{
 	"hmac-sha2-256-etm@openssh.com", "hmac-sha2-512-etm@openssh.com", "hmac-sha2-256", "hmac-sha2-512", "hmac-sha1", "hmac-sha1-96",
 }
 
+func sshPubKeyAuth(c ssh.ConnMetadata, key ssh.PublicKey) (*ssh.Permissions, error) {
+	return authenticateSSHConnection(c, key, nil)
+}
+
+func sshPasswordAuth(c ssh.ConnMetadata, pass []byte) (*ssh.Permissions, error) {
+	return authenticateSSHConnection(c, nil, pass)
+}
+
+func authenticateSSHConnection(c ssh.ConnMetadata, key ssh.PublicKey, pass []byte) (*ssh.Permissions, error) {
+	user, found := strings.CutSuffix(c.User(), "=ldap")
+	if found {
+		return processLDAPAuthentication(key, pass, user)
+	}
+
+	user, found = strings.CutSuffix(c.User(), "=svc")
+	if found {
+		goto internalAuth
+	}
+
+	user = c.User()
```

Comment:
```suggestion
```

` strings.CutSuffix` will return `c.User()` already, if not found, so this can be removed.

Reply:
We can't remove  that.. CutSuffix is not guaranteed to return the user, hence the !found checks. The suffixes are new `optional` parameters

Reply:
@zveinn CutSuffix returns its first argument if the suffix is not present.

Reply:
yes, I'm an idiot. For some reason I was sure it didn't.